### DISCUSSION TITLE
How to preserve Exif data

### DISCUSSION THREAD

**Discussion by bluegene37** [hm21/pro_image_editor#238] [Q&A] [answered]
Hi Guys,

I need your help.

How can i preserve exif data after saving an edited image?

This is my code:

      MaterialPageRoute(
        builder: (context) => ProImageEditor.memory(
          imageData,
          callbacks: ProImageEditorCallbacks(
            onImageEditingComplete: (Uint8List bytes) async {
               //  this is where i save the image after edit 
            },
          ),
        ),
      ),

**Comment by hm21** [comment]
I'm not sure why you want to change the exif data, but it should be possible with the [image](https://pub.dev/packages/image) package, which I also use in the image editor. You can see an example of how to do this [here](https://github.com/brendan-duncan/image/blob/main/doc/exif.md).
If you need to know the information about the current transformation of the image, like rotation, you can add a key `final editorKey = GlobalKey<ProImageEditorState>()` to the editor and read it as below:


```dart
var key = editorKey.currentState!;
var configs = key.stateHistory[key.stateManager.position].transformConfigs;
```

**Reply by bluegene37** [reply]
Hi @hm21 , thank you for the reply. I don't wanna change the exif data, what i want is, to preserve the exif after editing. I found out that edited image lost it's exif data. Any idea how to keep exif data after editing the image? Thank you!

**Reply by hm21** [reply, marked as answer]
Currently, the editor cannot automatically keep the exif data when generating a new image. To transfer some or all of the exif data, you must extract it from the original image and manually add it to the new one. As mentioned above, you can use the image package to read and write exif data, as demonstrated in this [documentation](https://github.com/brendan-duncan/image/blob/main/doc/exif.md).

**Reply by bluegene37** [reply]
Thank you @hm21 for the reply. That's now clear. Appreciated it much. Thank you for your effort on the plugin also.
